CMakeLists.txt

```diff
@@ -1,11 +1,36 @@
 # CMake4GDAL project is distributed under MIT license. See accompanying file LICENSE.txt.
+# Modified by Mi Zhang, LuoJiaNET Development and Research Group, Wuhan University, 2022
+
 cmake_minimum_required(VERSION 3.9...3.23)
 
 project(gdal LANGUAGES C CXX)
 include(CTest)
 
 set(GDAL_LIB_TARGET_NAME GDAL)
 
+###############################################################################
+# Generate setup.cfg
+###############################################################################
+if (WIN32)
+    set(GDAL_COMPILER mingw32)
```

Comment:
setting mingw32 for a MSVC build is not appropriate

Reply:
Adding the following code to if branch is optimal:
if (WIN32)
   if(MINGW)
       set(GDAL_COMPILER mingw32)
   else() #MSVC platform
       set(GDAL_COMPILER msvc)
   endif()
  endif()